astropy/table/column.py

```diff
@@ -383,6 +383,13 @@ def __new__(cls, data=None, name=None,
             if meta is None:
                 meta = data.info.meta
 
+        elif isinstance(data, list) and data == []:
+            # We should special case the empty list since np.array([], dtype)
+            # does not return an empty array with the requested dtype, and in
```

Comment:
I guess I'm behind the curve on this issue, but this statement confuses me.
```
In [2]: np.array([], dtype=np.int32)                                            
Out[2]: array([], dtype=int32)
```
Is that not an empty array with the requested dtype?

Reply:
@taldcroft it is "subarray" dtypes, which are a bit strange, e.g.:

```
In [4]: dt = np.dtype("(3)i,")                                                                                          

In [5]: dt.shape                                                                                                        
Out[5]: (3,)

In [6]: dt.ndim                                                                                                         
Out[6]: 1

In [7]: dt.base                                                                                                         
Out[7]: dtype('int32')

In [8]: np.array([], dtype=dt)                                                                                          
Out[8]: array([], shape=(0, 3), dtype=int32)
```

Reply:
OK, that makes sense. I think for this PR then we just need the comment to be even longer, e.g. " .. does not return an empty array with the requested dtype in certain cases like `dtype="(3)i,"` (see https://github.com/numpy/numpy/issues/17511).  In numpy 1.20-dev ...".